sklearn/cluster/affinity_propagation_.py

```diff
@@ -288,17 +295,21 @@ def fit(self, X, y=None):
         if self.affinity == "precomputed":
             self.affinity_matrix_ = X
         elif self.affinity == "euclidean":
-            self.affinity_matrix_ = -euclidean_distances(X, squared=True)
+            self.affinity_matrix_ = -euclidean_distances(X, squared=False)
         else:
             raise ValueError("Affinity must be 'precomputed' or "
                              "'euclidean'. Got %s instead"
                              % str(self.affinity))
 
-        self.cluster_centers_indices_, self.labels_, self.n_iter_ = \
-            affinity_propagation(
-                self.affinity_matrix_, self.preference, max_iter=self.max_iter,
-                convergence_iter=self.convergence_iter, damping=self.damping,
-                copy=self.copy, verbose=self.verbose, return_n_iter=True)
+        with warnings.catch_warnings():
+            warnings.simplefilter('always', ConvergenceWarning)
```

Comment:
Why is this necessary here? I would rather not mess with the warnings machine in our fit code but rather only the tests.